README.md

```diff
@@ -48,6 +48,12 @@ Conventionally used when `func` returns another option.
 * `some(value).flatMap(func)` returns `func(value)`
 * `none.flatMap(func)` returns `none`
 
+### filter(func)
```

Comment:
Minor thing, but `func` should be surrounded by asterisks. Also, for `filter`, would calling the argument `predicate` be clearer than `func`?